source/common/router/router.cc

```diff
@@ -519,56 +519,49 @@ void Filter::onResponseTimeout() {
   ENVOY_STREAM_LOG(debug, "upstream timeout", *callbacks_);
   cluster_->stats().upstream_rq_timeout_.inc();
 
-  // It's possible to timeout during a retry backoff delay when we have no upstream request. In
-  // this case we fake a reset since onUpstreamReset() doesn't care.
+  // It's possible to timeout during a retry backoff delay when we have no upstream request.
   if (upstream_request_) {
     if (upstream_request_->upstream_host_) {
       upstream_request_->upstream_host_->stats().rq_timeout_.inc();
     }
     upstream_request_->resetStream();
   }
 
-  onUpstreamReset(UpstreamResetType::GlobalTimeout, absl::optional<Http::StreamResetReason>());
+  updateOutlierDetection(timeout_response_code_);
+  std::string body =
+      timeout_response_code_ == Http::Code::GatewayTimeout ? "upstream request timeout" : "";
```

Comment:
It might be good store these string values as constants so we don't have to allocate temporary strings whenever we hit this code path (especially since we just end up calling `c_str()` on the string anyways), but that kinda falls outside the scope of this PR so I don't think that's necessary to do right now

Reply:
leaving it for now

Reply:
You should be able to safely change the type here to `absl::string_view` as well as onUpstreamAbort parameter, and pass it directly to `sendLocalReply`.